### PR TITLE
Add Dropout

### PR DESCRIPTION
**Pull request by yan12125**
Based on the implementation in https://github.com/ENOT-AutoDL/onnx2torch/pull/51

Unlike PR #51, I don't use "feature dropout" like nn.Dropout2d or
F.dropout2d, as ONNX does not define them [1].

[1] https://github.com/onnx/onnx/issues/113

### CODE REVIEW

File: onnx2torch/node_converters/dropout.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        ratio: float = 0.5,
+        training_mode: Optional[torch.Tensor] = None,
+    ) -> torch.Tensor:
+
+        if training_mode is not None:
+            training = training_mode.item()
+        else:
+            training = False
+
+        return F.dropout(input_tensor, p=ratio, training=training)
```
[the remark this answers is not in the document]
Comment (yan12125) [reply onnx2torch/node_converters/dropout.py:32]: Hmm, should I just ignore the training_mode parameter or report a warning/error when training_mode does not match self.training?

File: tests/node_converters/dropout_test.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        inputs_example=test_inputs,
+        opset_version=opset_version,
+    )
+
+    check_onnx_model(
+        model,
+        test_inputs,
+        atol_onnx_torch=10**-6,
+        atol_torch_cpu_cuda=10**-6,
+        atol_onnx_torch2onnx=10**-6,
```
[the remark this answers is not in the document]
Comment (yan12125) [reply tests/node_converters/dropout_test.py]: Thanks! Those are indeed unneeded. I removed them in the second commit.

File: onnx2torch/node_converters/dropout.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ivkalgin) [reply onnx2torch/node_converters/dropout.py:32]: You should ignore training attribute from onnx and do not report any warning/error.

[the remark this answers is not in the document]
Comment (yan12125) [reply onnx2torch/node_converters/dropout.py]: Sounds good, changed in the third commit.

[the remark this answers is not in the document]
Comment (senysenyseny16) [reply onnx2torch/node_converters/dropout.py]: Hi, Yen!

Thanks for your PR.

Your dropout implementation is valid except tests, on the other hand our tests system is not designed for "random" ops.
We will rewrite the tests later.

**Review by senysenyseny16 — APPROVED** [review]

File: onnx2torch/node_converters/dropout.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (yan12125) [reply onnx2torch/node_converters/dropout.py]: Thanks! Honestly, I'm not quite sure how to properly test the operator, either.
